src/riot-rs-threads/src/lib.rs

```diff
@@ -226,14 +369,20 @@ pub fn thread_create<T: Arguable + Send>(
     arg: T,
     stack: &'static mut [u8],
     prio: u8,
+    core_affinity: Option<CoreAffinity>,
 ) -> ThreadId {
     let arg = arg.into_arg();
-    unsafe { thread_create_raw(func as usize, arg, stack, prio) }
+    unsafe { thread_create_raw(func as usize, arg, stack, prio, core_affinity) }
 }
 
 /// Low-level function to create a thread without argument
-pub fn thread_create_noarg(func: fn(), stack: &'static mut [u8], prio: u8) -> ThreadId {
-    unsafe { thread_create_raw(func as usize, 0, stack, prio) }
+pub fn thread_create_noarg(
+    func: fn(),
+    stack: &'static mut [u8],
+    prio: u8,
+    core_affinity: Option<CoreAffinity>,
```

Comment:
Should `CoreAffinity` really awlays be part of the public API? Or instead have a `_with_affinity` version of each of these 3 thread-creation function?

Reply:
fine as is!